File: java/src/main/java/commonclasses/ListNode.java
/**
 * Definition for singly-linked list.
 */
package commonclasses;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        this.val = x;
    }

    public ListNode(int x, ListNode next) {
        this.val = x;
        this.next = next;
    }
}

File: golang/commonstructs/list_node.go
package commonstructs

// Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func (node1 *ListNode) Equals(node2 *ListNode) bool {
	if node1 == nil && node2 == nil {
		return true
	}

	if node1 == nil || node2 == nil {
		return false
	}

	if node1.Val != node2.Val {
		return false
	}

	return node1.Next.Equals(node2.Next)
}
